cypress/e2e/dataset-page.cy.ts

```diff
@@ -0,0 +1,98 @@
+import ProjectLinks from "../../src/app/project-links";
+
+describe("Dataset page", () => {
+    beforeEach(() => {
+        cy.intercept("POST", "/graphql").as("graphqlRequest");
+        cy.visit(Cypress.env("initDatasetName") as string);
+        cy.wait("@graphqlRequest");
+    });
+
+    it("Visits dataset component page", () => {
+        cy.url().should("include", Cypress.env("initDatasetName") as string);
+        cy.url().should("not.include", "?tab=");
+        cy.contains(
+            "Confirmed positive cases of COVID-19 in British Columbia",
+        ).should("be.visible");
+    });
+
+    it("Check switch tabs", () => {
+        ["data", "metadata", "history", "lineage"].forEach(
+            (tab: string, index: number) => {
+                cy.get("mat-button-toggle-group")
+                    .find("button")
+                    .eq(index + 1)
+                    .click();
+                cy.url().should("include", `?tab=${tab}`);
+            },
+        );
+    });
+
+    it("Should show page 404 when url include wrong name dataset and redirect to home page", () => {
+        cy.visit("/kamu/british-columbia-error.case-details");
```

Comment:
I recommend creating a file with test data for each component and using variables, so it's less likely to make mistakes and easier in the future